instruments/minghe/mhs5200a.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Provides the support for the MingHe low-cost function generator.
+
+Class originally contributed by Catherine Holloway.
+"""
+
+# IMPORTS #####################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+from builtins import range
+from enum import Enum
+
+import quantities as pq
+
+from instruments.abstract_instruments import Instrument
+from instruments.util_fns import ProxyList, assume_units
+
+# CLASSES #####################################################################
+
+
+class MHS5200(Instrument):
+    """
+    The MHS5200 is a low-cost, 2 channel function generator.
+
+    There is no user manual, but Al Williams has reverse-engineered the
+    communications protocol:
+    https://github.com/wd5gnr/mhs5200a/blob/master/MHS5200AProtocol.pdf
+    """
+    # pylint: disable=unused-variable
+    def __init__(self, filelike):
+        super(MHS5200, self).__init__(filelike)
+        self._channel_count = 2
+
+    # INNER CLASSES #
+
+    class Channel(object):
+        """
+        Class representing a channel on the MHS52000.
+        """
+
+        __CHANNEL_NAMES = {
+            1: '1',
+            2: '2'
+        }
+
+        def __init__(self, mhs, idx):
+            self._mhs = mhs
+            # Use zero-based indexing for the external API, but one-based
+            # for talking to the instrument.
+            self._idx = idx + 1
+            self._chan = self.__CHANNEL_NAMES[self._idx]
+            self._count = 0
+
+        @property
+        def amplitude(self):
+            """
+            Gets/Sets the amplitude of this channel.
+
+            :units: As specified (if a `~quantities.Quantity`) or assumed to be
+            of units volt.
+            :type: `~quantities.Quantity`
+            """
+            query = ":r{0}a".format(self._chan)
+            response = self._mhs.query(query)
+            return float(response.replace(query, ""))/100.0*pq.V
+
+        @amplitude.setter
+        def amplitude(self, new_val):
+            new_val = 100*assume_units(new_val, pq.V).rescale(pq.V).magnitude
```

Comment:
Why do we need to multiply by 100? What a strange number system this instrument uses.